packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -676,6 +680,51 @@ export function propControllerHandlesMiddleware(): Middleware<Dispatch, State, D
     }
 }
 
+export function elementTreeMiddleware(): Middleware<Dispatch, State, Dispatch> {
+  return ({ dispatch, getState }: MiddlewareAPI<Dispatch, State>) =>
+    (next: ReduxDispatch<Action>) => {
+      return (action: Action): Action => {
+        switch (action.type) {
+          case ActionTypes.REGISTER_DOCUMENT:
```

Comment:
Here and below, intercept document updates and dispatch corresponding element tree synchronization actions.